cycles-ledger/tests/tests.rs

```diff
@@ -156,46 +156,299 @@ fn install_fake_cmc(env: &StateMachine) {
 pub fn account(owner: u64, subaccount: Option<u64>) -> Account {
     Account {
         owner: Principal::from_slice(owner.to_be_bytes().as_slice()),
-        subaccount: subaccount
-            .map(|subaccount| subaccount.to_be_bytes().as_slice().try_into().unwrap()),
+        subaccount: subaccount.map(|subaccount| {
+            let mut subaccount_bytes = [0u8; 32];
+            subaccount_bytes[24..32].copy_from_slice(subaccount.to_be_bytes().as_slice());
+            subaccount_bytes
+        }),
+    }
+}
+
+struct TestEnv {
+    pub state_machine: StateMachine,
+    pub ledger_id: Principal,
+    pub depositor_id: Principal,
+}
+
+impl TestEnv {
+    fn setup() -> Self {
+        let state_machine = new_state_machine();
+        let ledger_id = install_ledger(&state_machine);
+        let depositor_id = install_depositor(&state_machine, ledger_id);
+        Self {
+            state_machine,
+            ledger_id,
+            depositor_id,
+        }
+    }
+
+    fn setup_with_ledger_conf(conf: LedgerConfig) -> Self {
+        let state_machine = new_state_machine();
+        let ledger_id = install_ledger_with_conf(&state_machine, conf);
+        let depositor_id = install_depositor(&state_machine, ledger_id);
+        Self {
+            state_machine,
+            ledger_id,
+            depositor_id,
+        }
+    }
+
+    fn upgrade_ledger(&self, args: Option<UpgradeArgs>) -> Result<(), CallError> {
+        let arg = Encode!(&Some(LedgerArgs::Upgrade(args))).unwrap();
+        self.state_machine
+            .upgrade_canister(self.ledger_id, get_wasm("cycles-ledger"), arg, None)
+    }
+
+    fn deposit(&self, to: Account, amount: u128, memo: Option<Memo>) -> DepositResult {
+        client::deposit(&self.state_machine, self.depositor_id, to, amount, memo)
+    }
+
+    fn get_block(&self, block_index: Nat) -> Block {
+        client::get_block(&self.state_machine, self.ledger_id, block_index)
```

Comment:
Do you use `client::get_block` elsewhere too?
Otherwise, you could simply move the implementation here.

Reply:
I haven't switched to the `TestEnv` In the tests of `create_canister` yes so the implementation need to stay for now.